lib/_tls_wrap.js

```diff
@@ -382,34 +413,38 @@ TLSSocket.prototype._wrapHandle = function(wrap) {
   }
 
   // Wrap socket's handle
-  var context = options.secureContext ||
-                options.credentials ||
-                tls.createSecureContext(options);
-  res = tls_wrap.wrap(handle._externalStream,
-                      context.context,
-                      !!options.isServer);
+  const context = options.secureContext ||
+                  options.credentials ||
+                  tls.createSecureContext(options);
+  const res = tls_wrap.wrap(handle._externalStream,
+                            context.context,
+                            !!options.isServer);
   res._parent = handle;
   res._parentWrap = wrap;
   res._secureContext = context;
   res.reading = handle.reading;
+  this[kRes] = res;
   Object.defineProperty(handle, 'reading', {
-    get: function get() {
-      return res.reading;
+    get: () => {
+      return this[kRes].reading;
     },
-    set: function set(value) {
-      res.reading = value;
+    set: (value) => {
```

Comment:
When changing this, you can directly switch to `get() {` and `set(value) {`

Reply:
This needs to be an arrow function. (I'm trying to avoid referencing res so the unsetting can happen in a separate function and there's no leak.)